Assets/Plugin/Generators/GeneratorRemapOnDemand.cs

```diff
@@ -0,0 +1,78 @@
+using System.Collections.Generic;
+using TMPro;
+using UnityEngine;
+
+public class RemapOnDemand : IDMXGenerator
+{
+    public DMXChannel ToggleChannel = 0;
+    public DMXChannel RemapFromChannelStart = 0;
+    public DMXChannel RemapToChannelStart = 0;
+    public EquationNumber RemapChannelLength = 0;
+    public EquationNumber ActivationThreshold = 127;
+    
+    public void Construct() { }
+    public void Deconstruct() { }
+    public void GenerateDMX(ref List<byte> dmxData)
+    {
+        //ensure capacity
+        dmxData.EnsureCapacity(RemapToChannelStart + (int)RemapChannelLength);
+        dmxData.EnsureCapacity(RemapFromChannelStart + (int)RemapChannelLength);
+        
+        if (dmxData[ToggleChannel] > ActivationThreshold)
+        {
+            //copy the source channels to the target channel
+            for (int i = 0; i < RemapChannelLength; i++)
```

Comment:
this should probably use the built in copy functions of arrays/lists instead of a manual for loop